dockstore-client/src/main/java/io/dockstore/client/cli/Client.java

```diff
@@ -183,21 +183,112 @@ public static String getCurrentVersion(String installLocation) {
         return currentVersion;
     }
 
-    public static String openCurrentVersion(URL link, String info){
+    /**
+     * This method will get information based on the json file on the link to the current version
+     * However, it can only be the information outside "assets" (i.e "name","id","prerelease")
+     * @param link
+     * @param info
+     * @return
+     */
+    public static String getFromJSON(URL link, String info){
         ObjectMapper mapper = getObjectMapper();
         Map<String,Object> mapCur;
         try{
             mapCur = mapper.readValue(link,Map.class);
             return mapCur.get(info).toString();
 
         } catch(IOException e){
-            exceptionMessage(e,"cannot find "+info+" in "+link,IO_ERROR);
         }
         return "null";
     }
 
     /**
-     * Check if the date of the current is later or earlier than latest version
+     * This method will return a map consists of all the releases
+     * @return
+     */
+    public static List<Map<String, Object>> getAllReleases(){
+        URL url;
+        try {
+            ObjectMapper mapper = getObjectMapper();
+            url = new URL("https://api.github.com/repos/ga4gh/dockstore/releases");
+            List<Map<String, Object>> mapRel;
+            try{
+                TypeFactory typeFactory = mapper.getTypeFactory();
+                CollectionType ct = typeFactory.constructCollectionType(List.class, Map.class);
+                mapRel = mapper.readValue(url, ct);
+                return mapRel;
+            } catch(IOException e){
+            }
+
+        }catch(MalformedURLException e){
+        }
+        return null;
+    }
+
+    /**
+     * This method will return the latest unstable version
+     * @return
+     */
+    public static String getLatestUnstableVersion(){
+        List<Map<String,Object>> allReleases = getAllReleases();
+        Map<String, Object> map;
+        for(int i=0;i<allReleases.size();i++){
+            map = allReleases.get(i);
+            if(map.get("prerelease").toString().equals("true")){
+                return map.get("name").toString();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This method will decide which version to upgrade to depending on the command "stable","unstable", and "none"
+     * for testing 'UpgradeTestIT'
+     * @param upCommand
+     * @param current
+     * @param latestStable
+     * @param latestUnstable
+     * @return
+     */
+
+    public static String decideOutput(String upCommand, String current, String latestStable, String latestUnstable){
```

Comment:
It looks like this method is only used for testing, please move it into the testing class.